src/lex/mod.rs

```diff
@@ -1039,7 +1019,31 @@ impl LiteralToken {
         match self {
             LiteralToken::Int(i) => LiteralValue::Int(i),
             LiteralToken::UnsignedInt(u) => LiteralValue::UnsignedInt(u),
-            LiteralToken::Float(f) => LiteralValue::Float(f),
+            LiteralToken::Float(rcstr) => {
+                let buf = rcstr.as_str();
+                let hex = buf.starts_with("0x");
+                let buf = buf
+                    .trim_end_matches("f")
+                    .trim_end_matches("F")
+                    .trim_end_matches("l")
+                    .trim_end_matches("L"); // Ignored for compatibility reasons
```

Comment:
This does not have the same behavior. Previously `1.0fl` would be parsed as `[Float(1.0), Id("l")]`. Now it will be parsed as one float. This needs to stop the first time it sees one of the letters.

Unless this only works on input that's already valid? In that case it should be fine.

Reply:
This is operating on a string after it has already been lexed so it shouldn't be able to have more than one suffix, but I'll try to do it correctly. Unfortunately `strip_suffix` is nightly which is the best way I can see to do it.